beacon-chain/sync/pending_blobs_queue.go

```diff
@@ -0,0 +1,144 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed"
+	statefeed "github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed/state"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
+	eth "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+)
+
+// processPendingBlobs listens for state changes and handles pending blobs.
+func (s *Service) processPendingBlobs() {
+	s.waitForChainStart()
+
+	eventFeed := make(chan *feed.Event, 1)
+	sub := s.cfg.stateNotifier.StateFeed().Subscribe(eventFeed)
+	defer sub.Unsubscribe()
+
+	// Initialize the cleanup ticker
+	cleanupTicker := slots.NewSlotTicker(s.cfg.chain.GenesisTime(), params.BeaconConfig().SecondsPerSlot/2)
```

Comment:
is there any docs on why it's triggering ever 500ms?
also many other processpending functions use async.RunEvery(
any reason for using the ticker here instead?

Reply:
It's triggered every 6s, I don't think it matters between `RunEvery` and `Ticket`. As long as it's pruned every 6s, some variance is ok. As for why 6s, it seems like a good compromise between running too often and not enough

Reply:
I would use a different ticker perhaps, running more routines at second zero is a bad idea, there are two different versions of tickers with offsets or with prescribed triggers within the slot

Reply:
I did a lot of work to decouple sync from depending on the blockchain package for genesis time - this PR re-introduces that dependency. Can you use `s.cfg.clock.GenesisTime` instead?